packages/ui/src/hooks/use-timeout.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState, useRef, useCallback } from 'react';
```

Comment:
Can you give an example of how this would be used?

Reply:
```tsx

   const { stopTimeout, startTimeout } = useTimeout({
      duration: 10000,
      onTimeout: () => console.log('ran after 10 seconds'),
    });

...

<Box 
  onMouseEnter={stopTimeout}
  onMouseLeave={startTimeout}>
    Something cool
</Box>
```

Reply:
used here to stop the remove fn from being called if the user is hovering on the toast: https://github.com/blockstack/ux/pull/375/files#diff-2baf28208398bac56df2c6fe898d2632R199-R202